### PR TITLE
feat: when validating OptimizationValidationMixin, fsdp_version shouldn't be flag as an warning

### PR DESCRIPTION
**Pull request by SamuelLarkin**
With the following config.yaml

```yaml
fsdp_version: 2
fsdp_config:
  offload_params: false
  state_dict_type: FULL_STATE_DICT
  auto_wrap_policy: TRANSFORMER_BASED_WRAP
  transformer_layer_cls_to_wrap: LlamaDecoderLayer
  reshard_after_forward: true
```

We get the following WARNING
```
[2026-06-08 12:08:41,580] [WARNING] [axolotl.utils.schemas.validation] Configuring FSDP fields with the `fsdp_` prefix is deprecated. Please omit the `fsdp_` prefix from the any fields in `fsdp_config`.
```

Where 
```
key == "fsdp_version"
fsdp_config={'offload_params': False, 'state_dict_type': 'FULL_STATE_DICT', 'auto_wrap_policy': 'TRANSFORMER_BASED_WRAP', 'transformer_layer_cls_to_wrap': 'LlamaDecoderLayer', 'reshard_after_forward': True, 'fsdp_version': 2}
```

`fsdp_version` is part of [FSDPConfig](https://github.com/axolotl-ai-cloud/axolotl/blob/main/src/axolotl/utils/schemas/fsdp.py#L15)

## Bug Fixes
* Resolved an issue where `fsdp_version` in FSDP configurations was incorrectly treated as deprecated and rewritten during configuration normalization.

### CODE REVIEW

**Review by winglian — APPROVED** [review]
good catch. thanks
